engine.py

```diff
@@ -198,6 +209,113 @@ def log_warning(self, msg):
+        :param bundle: The app, engine or framework object that is associated
+            with this window 
+        :param widget_class: The class of the UI to be
+        constructed. This must derive from QWidget.
+        
+        Additional parameters specified will be passed through to the
+        widget_class constructor.
+        """
+
+        # check to see if we just need to return the widget itself. Since we
+        # don't really have information about the panel outside of this call,
+        # we use a special flag to know when the info is needed and return it.
+        if hasattr(self, '_panel_info_request') and self._panel_info_request:
+            return {
+                'id': panel_id,
+                'title': title,
+                'bundle': bundle,
+                'widget_class': widget_class,
+                'args': args,
+                'kwargs': kwargs,
+            }
+
+        # try to locate the pane in the desktop and make it the current tab. 
+        for pane_tab in hou.ui.curDesktop().paneTabs():
+            if pane_tab.name() == panel_id:
+                pane_tab.setIsCurrentTab()
+                return
+
+        # panel support differs between 14/15. 
+        if self._panels_supported():
+
+            # if it can't be located, try to create a new tab and set the
+            # interface.
+            panel_interface = None
+            try:
+                for interface in hou.pypanel.interfacesInFile(self._panels_file):
+                    if interface.name() == panel_id:
+                        panel_interface = interface
+                        break
+            except hou.OperationFailed:
+                # likely due to panels file not being a valid file, missing, etc. 
+                # hopefully not the case, but try to continue gracefully.
+                hou.log_warning(
+                    "Unable to find interface for panel '%s' in file: %s" % 
+                    (panel_id, self._panels_file))
+
+            if panel_interface:
+                # the options to create a named panel on the far right of the
+                # UI doesn't seem to be present in python. so hscript it is!
+                hou.hscript("pane -S -m pythonpanel -o -n %s" % panel_id)
+                panel = hou.ui.curDesktop().findPaneTab(panel_id)
+
+                # different calls to set the python panel interface in Houdini
+                # 14/15
+                if hou.applicationVersion()[0] >= 15:
+                    panel.setActiveInterface(panel_interface)
+                else:
+                    # if SESI puts in a fix for setInterface, then panels
+                    # will work for houini 14. will just need to update
+                    # _panels_supported() to add the proper version. and 
+                    # remove this comment. 
+                    panel.setInterface(panel_interface)
+
+                panel.showToolbar(False)
+
+                return
```

Comment:
It may be possible that panels are supported but the panel interface couldn't be found in the installed panels file. So the `show_dialog` is the fallback for that scenario and panels not being supported at all. So I did this instead of duplicating the call to `show_panels` in an `else`. Does that make sense?